### ISSUE TITLE
Turn screen off doesn't work

### ISSUE THREAD

**Issue by lead-legs** [osmandapp/Osmand#7315]
Turn screen off no longer works - Since the update, I can set the screen to turn on automatically but no matter what seconds I set the screen doesn't tun off. #7315

**Comment by vshcherb** [comment]
The issue could be related on some specific Android settings like enabled FaceLock or system doesn't want to turn screen off.

**Comment by thehill383** [comment]
I have seen that in the latest update the functionality for "turn screen on" is restored. I haven't been able to test this as yet.
The real issue for me is for the main map screen to prevent the pin-lock from being activated. You should be able to use the power button to turn the screen on and off without encountering the pin/pattern lock. I have checked a few other apps and i am certain they have this functionality without needing to be in the device administrators group, which i understand was the issue. I asked the developer of ghostracer, how he does it (https://play.google.com/store/apps/details?id=com.bravetheskies.ghostracer&gl=GB) and this was his response"Anyway I have the option to remove the lock screen in settings. It's not been touched for years so might be out of date, but that adds two 'flags' to the 'window' FLAG_DISMISS_KEYGUARD and FLAG_SHOW_WHEN_LOCKED"
i am sure a number of apps are able to do this without being in the device administrators group, i think Strava is another

**Comment by lead-legs** [comment]
My settings are exactly the same as they were previously when this worked in Osmand.
My device is only used for Osmand navigating by bike. There is no face lock and other than Osmand not having the correct permissions, there's nothing stopping Osmand tithing the screen off.

**Comment by AndyGHenderson** [comment]
I have this problem. I do not have a lock screen of any kind active. The phone always starts directly to the menu (so I don't have to unlock each time I want to see the map while on my bike).

Also, screen fades and turns off automatically in normal use. It's only in OsmAnd that the screen stays on permanently until I press the power button to send the phone to sleep. 

As lead-legs says, my phone settings haven't changed since the automatic switch on/off worked before, and there hasn't been an OS update since either.

Using Motorola G3 with Android 6.0.1

Andy

**Comment by jsarenik** [comment]
Just a reference (the link currently at https://github.com/osmandapp/Osmand/issues/6675#issuecomment-516157223 is broken):
 https://github.com/osmandapp/Osmand/issues/6675

**Comment by Policjant** [comment]
I have the same problem on my navigation phone with android 5.0.1, on the old osmand version everything was working fine

**Comment by florian29** [comment]
Perhaps the idea of the developers is only to turn the screen on. But turning the screen off again, like it was the behaviour on version 3.2.7 , that is not mentioned. Obviously if this is the idea, than this feature is useless.

**Comment by jsarenik** [comment]
There is an app [Mapy.cz](https://play.google.com/store/apps/details?id=cz.seznam.mapy) which on pressing power button turns off the display and on pressing it again shows the map without locking the screen (the screenlock appears only after pressing the home button). I remember it worked like this in other map/navigation apps like [MapFactor](https://play.google.com/store/apps/details?id=com.mapfactor.navigator) or [Sygic](https://play.google.com/store/apps/details?id=com.sygic.aura) or [Offline Maps & Navigation](https://play.google.com/store/apps/details?id=com.navigation.offlinemaps.gps).

**Comment by max-klaus** [comment]
Could someone record a video with the issue? Thank you!

**Comment by jsarenik** [comment]
> Could someone record a video with the issue? Thank you!

Sure. https://youtu.be/vn52IQvZKEk

**Comment by jsarenik** [comment]
Now an example without issue (using Mapy.cz) https://youtu.be/mqOmi5W8Cq4

**Comment by AndyGHenderson** [comment]
jsarenik's video demonstrates an issue with the lock screen but the OP's issue is that after the screen comes on automatically, it never goes off again. I don't have the means to show a recording, unfortunately (I tried installing AZ recorder but Android kept complaining about overlays and I didn't get anywhere) but there really isn't much to show. The screen comes on automatically with the voice announcement but it never goes off again.

Definitely using the right profile with a setting of 5 seconds.

Andy

**Comment by max-klaus** [comment]
While device isn't going off - what actually is displaying on the screen, OsmAnd or idle screen?

**Comment by jsarenik** [comment]
@AndyGHenderson oops, I think I have hijacked this issue for something that bothers me. Sorry for that.

**Comment by max-klaus** [comment]
@jsarenik Thank you for the video. We will try to implement such case as well.

**Comment by jsarenik** [comment]
@max-klaus you are welcome. For the sake of tracking, do you want me to make or reopen another issue? Thank you!

**Comment by AndyGHenderson** [comment]
> While device isn't going off - what actually is displaying on the screen, OsmAnd or idle screen?

OsmAnd. The OsmAnd screen comes on automatically and never goes off.

Andy

**Comment by max-klaus** [comment]
> @max-klaus you are welcome. For the sake of tracking, do you want me to make or reopen another issue? Thank you!

no, 7315 is enough.

**Comment by apfelstrudel140** [comment]
I have the same problem. OsmAnd disables the screen to turn off. When pressing the powerbutton it is off for a second and after that it turns on again. I also tried to disable the auto screen lock and changes the settings for the power button, but it doesn't work.

Oneplus x, android 6.0.1, OsmAnd+ 3.4.4

**Comment by DreadPirateRoberts-1000** [comment]
Same problem here.  It appears to be related to the "Proximity sensor" option.  If I start a navigation, select a 'Time' in the 'turn screen on' section the power button turns off the screen.
If I also check 'Proximity Sensor' in the navigation options, pressing the power button turns off the screen but it turns on again within 1 sec.  This remains the case until I exit Osmand+.

This is seen sitting at my desk.  I'll check behaviour on the bike later.

Steps:
Start Osmand+
Start navigation - choose destination - press 'Start'
Lock screen: OK
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - choose 'Time' - set '10 seconds' - return and press 'Continue'
Lock screen: OK
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - check 'Proximity sensor' - return and press 'Continue'
Lock screen: Screen turns on within 1 sec
Stop navigation, exit Osmand+, restart Osmand+ and lock screen works correctly again.



Cubot J3, android 'Go' 8.1.0, Osmand+ 3.4.5

**Comment by AndyGHenderson** [comment]
FWIW, without a lock screen, the screen stays on after coming on automatically with - or without - the proximity sensor option.

Andy

**Comment by Posteinwurf** [comment]
same issue on my device
screen turns on but never off.
worked with older osmand version.

**Comment by DreadPirateRoberts-1000** [comment]
> Same problem here. It appears to be related to the "Proximity sensor" option. If I start a navigation, select a 'Time' in the 'turn screen on' section the power button turns off the screen.
> If I also check 'Proximity Sensor' in the navigation options, pressing the power button turns off the screen but it turns on again within 1 sec. This remains the case until I exit Osmand+.
> 
> This is seen sitting at my desk. I'll check behaviour on the bike later.
> 
> Steps:
> Start Osmand+
> Start navigation - choose destination - press 'Start'
> Lock screen: OK
> Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - choose 'Time' - set '10 seconds' - return and press 'Continue'
> Lock screen: OK
> Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - check 'Proximity sensor' - return and press 'Continue'
> Lock screen: Screen turns on within 1 sec
> Stop navigation, exit Osmand+, restart Osmand+ and lock screen works correctly again.
> 
> Cubot J3, android 'Go' 8.1.0, Osmand+ 3.4.5

Further testing on the bike this time:
Test 1:
Start Osmand+, start navigation, press 'Start'
Lock screen: OK
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - choose 'Time' - set '10 seconds' - return and press 'Continue'
Lock screen: OK - Osmand+ turns screen on and off appropriately during navigation.
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - check 'Proximity sensor' - return and press 'Continue'
Lock screen: Screen turns on within 1 sec
Stop navigation, exit Osmand+, restart Osmand+ - choose navigation - choose 'Options' - choose 'Navigation settings' - uncheck 'Proximity sensor' - return and lock screen works correctly again.

Test 2:
Start Osmand+, start navigation, press 'Start'
Lock screen: OK
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - check 'Proximity sensor' - choose 'Time' - set 'Never' - return and press 'Continue'
Lock screen: OK - Osmand+ does not respond to proximity sensor
Stop navigation, exit Osmand+, restart Osmand+

Test 3 - Not on bike, just at desk:
In Settings - Security - Lock screen - choose 'None'
Start Osmand+, start navigation, press 'Start'
Lock screen: OK
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - choose 'Time' - set '10 seconds' - return and press 'Continue'
Lock screen: OK
Unlock screen - choose navigation - choose 'Options' - choose 'Navigation settings' - check 'Proximity sensor' - return and press 'Continue'
Lock screen: Screen turns on within 1 sec
Stop navigation, exit Osmand+, restart Osmand+ - choose navigation - choose 'Options' - choose 'Navigation settings' - uncheck 'Proximity sensor' - return
In Settings - Security - Lock screen - choose 'Swipe'
Lock screen works correctly again.

Test 4 - Back on bike, this time with a usb charger connected to dynamo:
* When phone is plugged into usb charger, it will charge or not depending on road speed. This causes the screen to light up at each change from charge to no charge. *
Start Osmand+, start navigation, press 'Start' (Screen on Time is already set to 10 sec)
Lock screen: OK - Osmand+ turns screen on and off appropriately during navigation.
When usb charger toggles between charge and no charge, the lock screen lights up displaying Osmand+ notification but does not display the map. This screen times out with the phone display settings. This is correct behaviour.

Conclusion:
Proximity sensor checked along with a screen on time set causes Osmand+ to light the screen constantly, even after using power button to blank screen. Presumably this is down to the sensor polling?

**Comment by florian29** [comment]
The same issue with OsmAnd 3.4.6. The screen turns on but doesn't turn off anymore.

**Comment by DreadPirateRoberts-1000** [comment]
> FWIW, without a lock screen, the screen stays on after coming on automatically with - or without - the proximity sensor option.
> 
> Andy

Confirming this also for me.

**Comment by Sixtron** [comment]
Osmand+ V3.4.6
Android 9, Emui 9.0.0
Cellphone: Honor play

Same problem. Screen turn on, but not off.

I am not using the proximity switch.

I am using no screenlock.

Osmand don't ask for administration when installing. I can not add Osmand as admin manually with my cellphone.

Uninstall and reinstall don't help.

**Comment by DreadPirateRoberts-1000** [comment]
> Osmand+ V3.4.6
> Android 9, Emui 9.0.0
> Cellphone: Honor play
> 
> Same problem. Screen turn on, but not off.
> 
> I am not using the proximity switch.
> 
> I am using no screenlock.
> 
> Osmand don't ask for administration when installing. I can not add Osmand as admin manually with my cellphone.
> 
> Uninstall and reinstall don't help.

Try setting a 'swipe' screenlock.  
Works correctly for me with proximity switch: Off, screenlock: Swipe.

**Comment by florian29** [comment]
> Try setting a 'swipe' screenlock.
> Works correctly for me with proximity switch: Off, screenlock: Swipe.

Having to swipe the screen or something like this is not useful and not allowed while driving.

**Comment by max-klaus** [comment]
We have introduced fix for the issue. However it not released yet. You may try it using Nightly Build of the OsmAnd here: http://osmand.net/downloads

**Comment by AndyGHenderson** [comment]
> We have introduced fix for the issue. However it not released yet. You may try it using Nightly Build of the OsmAnd here: http://osmand.net/downloads

Just gave it a try. It's a step forward as the screen now goes off after coming on automatically but it seems to use the Android 'Sleep after inactivity' setting (in my case 1 minute) not the OsmAnd setting (in my case 5 seconds).

Even if I set the Android sleep timeout to its minimum value (15 seconds) it would still consume more battery than if it used the OsmAnd value.

Andy
Android 6.0.1 on a Motorola G3.

**Comment by AndyGHenderson** [comment]
> > Osmand+ V3.4.6
> > Android 9, Emui 9.0.0
> > Cellphone: Honor play
> > Same problem. Screen turn on, but not off.
> > I am not using the proximity switch.
> > I am using no screenlock.
> > Osmand don't ask for administration when installing. I can not add Osmand as admin manually with my cellphone.
> > Uninstall and reinstall don't help.
> 
> Try setting a 'swipe' screenlock.
> Works correctly for me with proximity switch: Off, screenlock: Swipe.

With a swipe unlock, the screen comes on automatically without needing the swipe but no change for me. It still doesn't go off automatically. Could be because I'm using an older version of Android.

Andy
Android 6.0.1 on Motorola G3

**Comment by Sixtron** [comment]
This fixed version (7/2019) seems to be older then the latest release (8/2019) ?

**Comment by max-klaus** [comment]
The fix does not restore functionality as it was before (with turn screen off right away) since it require Admin permission. Such approach cannot be used anymore. So current solution is the only possible way to ask android to turn off the screen. Unfortunately it cannot be done immediately.

**Comment by max-klaus** [comment]
@Sixtron Build number should be #24715
http://download.osmand.net/latest-night-build/OsmAnd-default.apk

**Comment by AndyGHenderson** [comment]
> @Sixtron Build number should be #24715
> http://download.osmand.net/latest-night-build/OsmAnd-default.apk

I got 24714d released 2019-08-09

Andy

**Comment by Sixtron** [comment]
Why not use admin rights to turn off the screen like before ? Many apps still use it.
You say, the screen will not shutdown because Osmand give the order ?
The screen will turn of because the system screen timeout ?

**Comment by AndyGHenderson** [comment]
> The fix does not restore functionality as it was before (with turn screen off right away) since it require Admin permission. Such approach cannot be used anymore. So current solution is the only possible way to ask android to turn off the screen. Unfortunately it cannot be done immediately.

I'm puzzled. By providing an option to tell OsmAnd how long before it should turn off the screen, you are creating an expectation that it will turn off after that period of time. The above suggests, however, that it will not work and that - regardless of the OsmAnd setting - the screen will obey only the Android screen timeout setting.

Did I get that right? If so, the new setting would be better as a simple 'Turn screen on automatically' yes/no.

**Comment by max-klaus** [comment]
We decided to not use the admin rights because it could cause kick app off the market which was happening already.
As I can see - behaviour of devices differs when app try to lock the screen new way. Some waits for system lock time, some goes to idle screen after period of time defined in the OsmAnd for few seconds and then turn off the screen.

**Comment by Sixtron** [comment]
And can you fix it that the screen goes off after the period of time set in Osmand ?

**Comment by max-klaus** [comment]
It is already done, but don't work on all devices.

**Comment by Sixtron** [comment]
Ok, can you make it work on all devices ?
If not, can you maybe make an option to give admin rights for the cause that it's not working?

**Comment by max-klaus** [comment]
No, because it depends on the device and not - we are not planning to return device permission to the app.

**Comment by florian29** [comment]
> We decided to not use the admin rights because it could cause kick app off the market which was
> happening already.
How about making at least a version with admin rights for F-Droid? For the google players you could make an add on with this admin rights and the add on could be downloaded outside the market.

**Comment by Themanwithoutaplan** [comment]
Does it make sense to provide debug logs for this? I had all kinds of fun with this combined with facelock on a recent tour: disabling auto-unlock for facelock seems to be the best solution but OsmAnd still generally wouldn't unlock. I can imagine there might be a conflict there. Seeing as I normally have SmartLock set for my headset this isn't really an issue but it's important to know what should work.

In case it's any use I can test the same scenarios with an S5 (klte) and S10e.

**Comment by Sixtron** [comment]
I am just on the trip with my bicycle and testing the Osmand nighty. Screen just turn on with no password set and screen don't turn off anyway what settings i choose.
When i set 15 seconds in the system to turn the screen off, then it works in the system but not when using Osmand Nighty on top of the screen.

**Comment by apfelstrudel140** [comment]
> @Sixtron Build number should be #24715
> http://download.osmand.net/latest-night-build/OsmAnd-default.apk

I installed the night build. So far I have the same responses. The screen will go out for a second and then goes on again. I tried different kind of scenarios, normal navigation and by gpx file.

**Comment by Sixtron** [comment]
For me night build was working on my last bicycle trip.

**Comment by apfelstrudel140** [comment]
On a samsung phone (android 7.0) the screen will go on, but don't go off. This is with OsmAnd version 3.4.6 and a screen time-out of 15 seconds.

**Comment by benaxe** [comment]
I have 3.4.6 installed. If you leave voice on and mute the volume on the phone the screen switches off properly.

**Comment by swertz** [comment]
Note that the two issues discussed are linked:

1) Screen doesn't turn off automatically -> can still turn it off manually using the screen lock button?
2) Turning the screen back on using the lock button -> before it would just show OSMand again, now we have to unlock manually (and swiping or giving the PIN code is much less practical when riding a bike than just pushing the screen lock button).

So, solving 2) would perhaps help people who complain about 1)? Not to mention it would be nice to be able to turn the screen back on at any time without having to unlock, which was the behaviour a few months ago...

**Comment by Themanwithoutaplan** [comment]
With the point releases on the beta channel it would be great to know which bugs were fixed. Got 3.4.8 and was going to test it yesterday but before I even got round to it was hitting problems with OsmAnd opening without request, Smart Lock (headset) seems a bit trigger happy at the moment!

**Comment by apfelstrudel140** [comment]
> I have 3.4.6 installed. If you leave voice on and mute the volume on the phone the screen switches off properly.

This solved my problem. Screen turned on and off automatically now!
Thanks!

**Comment by AndyGHenderson** [comment]
People suggesting solutions: please specify which OS and which version you are using. The issue seems to depend on which version of Android you use so claiming a 'fix' is likely to confuse people - especially the devs.

Andy

**Comment by benaxe** [comment]
> I have 3.4.6 installed. If you leave voice on and mute the volume on the phone the screen switches off properly.

Not really a fix, workaround at its best but does the trick. Runs on Android 8.0.0

**Comment by florian29** [comment]
> 
> 
> I have 3.4.6 installed. If you leave voice on and mute the volume on the phone the screen switches off properly.

Doesn't work on Android 4.4.4.

**Comment by ibabbv** [comment]
Same issue on my older navigation device, screen turns on but NOT off. (However, will turn off after approx 30 sec. inactivity time out)
Did work with older OsmAnd+ version.  Now  OsmAnd 3.4.8,  Android 4.4.2  Samsung S4Mini (no more updates on Android)
Works fine on Samsung S10E,  Android 9   OsmAnd+ 3.4.8
So, to me  it seems related to Android version.
I want to keep using the S4Mini because it fits the holder on my racing bike.

**Comment by Sixtron** [comment]
On my last bicycle trip i used Osmand Nighty and Osmand +. By both the screen turned off, but before it turned off, i saw the unlock screen for some seconds.

Apart from this it was so terrible to navigate with Osmand because the screen don't turned on when you had to change the direction, just some seconds before. When turn, screen off.
Also all turning commands came to late, like lags in a game, even i had about 18 satellites for recieving. I was so angry to you deveopers.
Next thing, nearly every time when a new update comes, you have to make again your personal preferences in the app.
Can not be true.

**Comment by ibabbv** [comment]
Additional info: same problem  on other device:
Samsung tablet S2:   OsmAnd 3.4.8   Android 7.0     Screen turns on, but NOT off after selected time out (5sec). 
So far the only correct working device is my S10E with Android 9.

**Comment by AndyGHenderson** [comment]
OsmAnd 3.4.8 Android 6.0.1 on Motorola G3; no unlock screen.
No improvement. Screen comes on at right times, but never goes off.

**Comment by Themanwithoutaplan** [comment]
> Same issue on my older navigation device, screen turns on but NOT off. (However, will turn off after approx 30 sec. inactivity time out)
> Did work with older OsmAnd+ version. Now OsmAnd 3.4.8, Android 4.4.2 Samsung S4Mini (no more updates on Android)
> Works fine on Samsung S10E, Android 9 OsmAnd+ 3.4.8
> So, to me it seems related to Android version.
> I want to keep using the S4Mini because it fits the holder on my racing bike.
You should be able to flash a more recent version onto the S4 Mini, I'm sure had something else on mine before it got pinched. But, yes, the OS version does make a difference because Google significantly changed the way permissions were handled in Android 6 (I think it was this). I think things have changed again with Android 9, particularly if something like facial recognition is active.

Having only just recently switched from my S5 for a similar reason - having a good case and bike holder is far more important than the latest shiny, shiny - I would suggest you think about getting a case for the S10. I already had a quadlock (Ottercase + adapter for the S5), Quadlockcase for S10e. The newer screens are definitely more legible in bright sunlight and, more importantly where I live, they're less susceptible to water droplets. With the S5 if it was raining when the screen was on, the phone would think it was being operated on. None of those problems with the S10e, though, of course, I still had all the problems with the screenlock not switching off. I really think the only way to solve these is with a debug build.

**Comment by lead-legs** [comment]
This has now been fixed for me in both devices
Motorola G3 (Osprey) Android 7.1.2. Osmand 3.4.8 (2019-08-20)
Motorola E (Surnia) Android 7.1.2 Osmand 3.4.8 (2019-08-20)

**Comment by florian29** [comment]
Fixed on S4 mini Android 4.4.4 OsmAnd 3.4.8

**Comment by florian29** [comment]
with the new version 3.5.2 the issue came back for me on S4 mini Android 4.4.4: the screen doesn't turn on, with the version 3.4.8 it worked

**Comment by Themanwithoutaplan** [comment]
> with the new version 3.5.2 the issue came back for me on S4 mini Android 4.4.4: the screen doesn't turn on, with the version 3.4.8 it worked

Same here, though I've not had it work properly on the S10e at all.

**Comment by florian29** [comment]
> 
> 
> with the new version 3.5.2 the issue came back for me on S4 mini Android 4.4.4: the screen doesn't turn on, with the version 3.4.8 it worked

after a quick check with the version 3.5.3 it seems to be fixed for me

**Comment by omawnakw** [comment]
Not sure the issue is the same, but looks similar.
OsmAnd 3.5.5, Android 9, Samsung Galaxy Note 9 (SM-N960F)
Smart unlock and touch unlock enabled.
    When Osmand app is active it prevents the screen to turn off. Whether it just shows the map, any settings screen, in navigation mode or not - the screen never turns off automatically. 
    There is NO problem with screen locking or turning screen of manually using power/lock button.
If I turn off screen manually and then turn it on Osmand shows instead of the lock screen. It prompts password or unlock id when trying to switch to another app or open system settings etc - seems to work properly.
    In 3.5.5 I can't find the option to toggle OsmAnd appearing above the lock screen - was it removed?

**Comment by vshcherb** [comment]
Not to confuse with too many requests on same issue, I will open a different request for keep screen on option and close this one.

**Comment by EssBee59** [comment]
I had a similar problem on android 10...
not sure it is exactly the same, but you can have a look at my comment on #8034 (se  link above)
regards
Ess Bee

**Comment by ataman76** [comment]
Screen wake-up using the proximity sensor does not work on Samsung Galaxy S22 and Android 14.
When you try to manually unlock the screen, the Osmand navigation screen does not show, only the lock screen (biometric lock set using fingerprint)
